### PR TITLE
#20 Improve type definitions

### PR DESCRIPTION
**Pull request by mikheevm**
This PR:
- Setup eslint
- Improve type definitions
- Update dependencies to the latest version

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,9 +1,21 @@
 export type Networks = 'rinkeby' | 'mainnet';
 
+type ValueOf<T> = T[keyof T];
+
+export interface Transaction {
+  to: string;
+  value: string;
+  data?: string;
```
Comment (mikheevm) [inline src/index.ts:8]: @rmeissner @nicosampler should this be optional or required?

Comment (rmeissner) [reply src/index.ts:8]: Question is if the safe react side can handle it ;)

Comment (rmeissner) [reply src/index.ts:8]: If it can I am also fine with nullable

Comment (mikheevm) [reply src/index.ts:8]: No, it cannot. So the question is: should it handle it

Comment (rmeissner) [reply src/index.ts]: In the future probably yes, but not for this first version

**Pull request merged**